explore/src/main/scala/explore/Routing.scala

```diff
@@ -38,13 +39,13 @@ object Routing {
         .notFound(redirectToPage(HomePage)(SetRouteVia.HistoryPush))
         .verify(HomePage, ObsPage(Observation.Id.unsafeFromString("GS2020A-Q-1")))
         .onPostRenderP {
-          case (_, ObsPage(obsId), view) =>
-            view.zoomL(RootModel.obsId).set(Option(obsId)).runInCB *>
-              Callback.log(s"id:1 $obsId")
-          case _                         => Callback.empty
+          case (prev, next, view) if prev =!= next.some =>
+            Callback.log(s"Routing.onPostRender triggered [$prev] => [$next]") >>
+              view.zoomL(RootModelRouting.lens).set(next).runInCB
```

Comment:
That's the idea, but this will only trigger if the URL is changed manually, since we are updating it silently within the app.

It works fine to initialize the model on a newly loaded application. If the application is loaded and you change the URL manually though, it reloads. I don't know if it's possible to avoid this.